sunpy/instr/aia/response.py

```diff
@@ -285,36 +286,54 @@ def calculate_wavelength_response(self, channel, use_total_range=False, use_resp
 
         var = self.get_channel_data(channel)
 
-        if use_table2_gain:
-            self.calculate_effective_area(channel)
-            self.calculate_system_gain(channel, use_table2_gain=True)
-            self.wavelength_response = self.effective_area * self.system_gain * dn_per_photon
+        # # move to test
+        # if use_table2_gain:
+        #     self.calculate_effective_area(channel)
+        #     self.calculate_system_gain(channel, use_table2_gain=True)
+        #     self.wavelength_response = self.effective_area * self.system_gain * dn_per_photon
 
+        # # move to test
+        # elif use_response_table2:
+        #     # returns the instrument response values listed in Table 2
+        #     i = self.channel_list.index(channel)
+        #     response = [0.664, 1.785, 3.365, 1.217, 1.14, 0.041, 0.027, 0.0024, 0.0046][i]
+        #     self.wavelength_response = response
 
-        elif use_response_table2:
-            # returns the instrument response values listed in Table 2
-            i = self.channel_list.index(channel)
-            response = [0.664, 1.785, 3.365, 1.217, 1.14, 0.041, 0.027, 0.0024, 0.0046][i]
-            self.wavelength_response = response
+        # default is to use these values
+        if use_genx_values:
 
-        elif use_genx_values:
-
-            channel_index = self.wavelength_range_index(channel)
 
             self.calculate_system_gain(channel, use_genx_values=True)
-            if use_total_range:
-                self.wavelength_response = var['effective_area'] * (self.system_gain * dn_per_photon)
-            else:
-                self.wavelength_response = var['effective_area'][channel_index] * (
-                    self.system_gain * dn_per_photon)
+            wavelength_response = var['effective_area'] * (self.system_gain * dn_per_photon)
+
+
+            # if total range removed. float value returned, moving away from this!
+            # else:
+            #     self.wavelength_response = var['effective_area'][channel_index] * (
+            #         self.system_gain * dn_per_photon)
 
         else:
             self.calculate_system_gain(channel)
             self.calculate_effective_area(channel, total_range=True)
+            wavelength_response = self.effective_area * self.system_gain
+
 
-            self.wavelength_response = self.effective_area * self.system_gain
 
+        # wavelength range selected, conversion to index by multiplying by ten.
```

Comment:
This whole section (322-336) is not needed. The wavelength response functions already peak very strongly in wavelength space by their very definition. This is primarily because of the effective area (see Fig. 8, Boerner et al., 2012) as well as the reflectance (see Fig. 5, same paper).